### PR TITLE
[FLINK-34510][Runtime/State]Rename RestoreMode to RecoveryClaimMode

### PR DESCRIPTION
**Pull request by spoon-lz**
## What is the purpose of the change

As a subitem of FLIP-406,Rename RestoreMode to RecoveryClaimMode

## Brief change log

Change the `RestoreMode` class name and variable name to `RecoveryClaimMode`

## Verifying this change

This change is a minor rework and is already covered by existing tests.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes )
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (yes)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not documented)

### CODE REVIEW

**Comment by spoon-lz** [comment]
@flinkbot run azure

**Comment by spoon-lz** [comment]
@flinkbot run azure

**Comment by spoon-lz** [comment]
@flinkbot run azure

File: docs/layouts/shortcodes/generated/rest_v1_dispatcher.html [hunk 1]
```diff
@@ -571,7 +571,7 @@
     "allowNonRestoredState" : {
       "type" : "boolean"
     },
-    "claimMode" : {
+    "restoreMode" : {
```
Comment (Zakelly) [inline docs/layouts/shortcodes/generated/rest_v1_dispatcher.html:574]: Ah, don't know why is this changed? Caused by re-generating the documents?

Comment (spoon-lz) [reply docs/layouts/shortcodes/generated/rest_v1_dispatcher.html]: Sorry I made the mistake of manually modifying, after re-executing `generate-rest-docs`, `rest_v1_dispatcher.html`did not need to be modified, but' `est_v1_dispatcher.yml` did.

File: flink-runtime-web/src/test/resources/rest_api_v1.snapshot [hunk 2]
```diff
@@ -4190,4 +4190,4 @@
       }
     }
   } ]
-}
```
Comment (Zakelly) [inline flink-runtime-web/src/test/resources/rest_api_v1.snapshot:4193]: Ah, it's better to revert this change

**Review by Zakelly — COMMENTED** [review]
Overall LGTM. Thanks for the PR

File: flink-runtime-web/src/test/resources/rest_api_v1.snapshot (hunk 2 above)
Comment (spoon-lz) [reply flink-runtime-web/src/test/resources/rest_api_v1.snapshot]: This change was reverted, and the commits merge and rebase have been completed.

File: flink-runtime/src/main/java/org/apache/flink/runtime/jobgraph/SavepointRestoreSettings.java [hunk 3]
```diff
@@ -94,8 +96,8 @@ public boolean allowNonRestoredState() {
     }
 
     /** Tells how to restore from the given savepoint. */
-    public @Nonnull RestoreMode getRestoreMode() {
-        return restoreMode;
+    public @Nonnull RecoveryClaimMode getRestoreMode() {
```
Comment (fredia) [inline flink-runtime/src/main/java/org/apache/flink/runtime/jobgraph/SavepointRestoreSettings.java:99]:
```suggestion
    public @Nonnull RecoveryClaimMode getRecoveryClaimMode() {
```

File: flink-runtime-web/src/main/java/org/apache/flink/runtime/webmonitor/handlers/JarRunRequestBody.java [hunk 4]
```diff
@@ -135,12 +136,12 @@ public String getSavepointPath() {
 
     @Nullable
     @JsonIgnore
-    public RestoreMode getRestoreMode() {
-        return restoreMode == null ? deprecatedRestoreMode : restoreMode;
+    public RecoveryClaimMode getRestoreMode() {
```
Comment (fredia) [inline flink-runtime-web/src/main/java/org/apache/flink/runtime/webmonitor/handlers/JarRunRequestBody.java:139]:
```suggestion
    public RecoveryClaimMode getRecoveryClaimMode() {
```

**Review by fredia — APPROVED** [review]
Thanks for the PR, overall LGTM, little minor comments.

**Comment by spoon-lz** [comment]
@fredia Will need `StateBackend#supportsNoClaimRestoreMode` method name changed to `supportsNoRecoveryClaimMode`?

**Review by fredia — APPROVED** [review]

**Comment by fredia** [comment]
> @fredia Will need `StateBackend#supportsNoClaimRestoreMode` method name changed to `supportsNoRecoveryClaimMode`?

Thanks for the update, both are fine for me since `StateBackend#supportsNoClaimRestoreMode` returns a boolean value.
